src/commands/service/create-revision.ts

```diff
@@ -91,55 +58,133 @@ export const commandDecorator = (command: commander.Command): void => {
-      if (typeof description !== "string") {
-        throw Error(
-          `--description must be of type 'string', ${typeof description} given.`
-        );
-      }
-      if (typeof remoteUrl !== "string") {
-        throw Error(
-          `--remote-url must be of type 'string', ${typeof remoteUrl} given.`
-        );
-      }
-      if (typeof sourceBranch !== "string") {
-        throw Error(
-          `--source-branch must be of type 'string', ${typeof sourceBranch} given.`
-        );
-      }
-      if (typeof personalAccessToken !== "string") {
-        throw Error(
-          `--personal-access-token must be of type 'string', ${typeof personalAccessToken} given.`
-        );
-      }
-      if (typeof orgName !== "string") {
-        throw Error(
-          `--org-name must be of type 'string', ${typeof orgName} given.`
-        );
-      }
+      await makePullRequest(
+        defaultRings,
+        title,
+        sourceBranch,
+        description,
+        orgName,
+        remoteUrl,
+        personalAccessToken
+      );
 
-      ////////////////////////////////////////////////////////////////////////
-      // Main
-      ////////////////////////////////////////////////////////////////////////
-      // Make a PR against all default rings
-      for (const ring of defaultRings) {
-        if (typeof title !== "string") {
-          title = `[SPK] ${sourceBranch} => ${ring}`;
-          logger.info(`--title not set, defaulting to: '${title}'`);
-        }
-        await createPullRequest(title, sourceBranch, ring, {
-          description,
-          orgName,
-          originPushUrl: remoteUrl,
-          personalAccessToken
-        });
-      }
       await exitCmd(logger, process.exit, 0);
     } catch (err) {
       logger.error(err);
       await exitCmd(logger, process.exit, 1);
     }
   });
 };
+
+/**
+ * Gets the default rings
+ * @param targetBranch Target branch/ring to create a PR against
+ * @param bedrockConfig The bedrock configuration file
+ */
+export const getDefaultRings = (
+  targetBranch: string | undefined,
+  bedrockConfig: IBedrockFile
+): string[] => {
+  const defaultRings: string[] = targetBranch
+    ? [targetBranch]
+    : Object.entries(bedrockConfig.rings || {})
+        .map(([branch, config]) => ({ branch, ...config }))
+        .filter(ring => !!ring.isDefault)
+        .map(ring => ring.branch);
+  if (defaultRings.length === 0) {
+    throw Error(
+      `Default branches/rings must either be specified in ${join(
+        __dirname,
+        "bedrock.yaml"
+      )} or provided via --target-branch`
+    );
+  }
+  logger.info(
+    `Creating pull request against branches: ${defaultRings.join(", ")}`
+  );
+  return defaultRings;
+};
+
+/**
+ * Gets the source branch or parses git for the source branch
+ * @param sourceBranch The source branch
+ */
+export const getSourceBranch = async (
+  sourceBranch: string | undefined
+): Promise<string | undefined> => {
```

Comment:
base on the code, I do not see that this function will return undefined. Maybe I am wrong. Can you cross check? thanks